addons/website/views/website_templates.xml

```diff
@@ -111,13 +111,16 @@
             </t>
         </t>
         <!-- `alternate`/`canonical` mainly useful to crawlers/bots/SEO tools, which test the website as public user -->
-        <t t-if="request and request.is_frontend_multilang and website and website.is_public_user()">
+        <t t-if="request and request.is_frontend_multilang">
             <t t-set="alternate_languages" t-value="website._get_alternate_languages(canonical_params=canonical_params)"/>
-            <t t-foreach="alternate_languages" t-as="lg">
-        <link rel="alternate" t-att-hreflang="lg['hreflang']" t-att-href="lg['href']"/>
-            </t>
+            <t t-foreach="alternate_languages" t-as="lg" t-if="len(alternate_languages) > 1">
+        <link rel="alternate" t-att-hreflang="lg['url_code']" t-att-href="lg['href']"/>
+        <link rel="alternate" hreflang="x-default" t-att-href="lg['href']" t-if="lg['id'] == website._get_cached('default_lang_id')"/>
+        <link rel="canonical" t-att-href="lg['href']" t-if="lg['code']==lang"/>
+                </t><t t-else="">
+        <link rel="canonical" t-att-href="alternate_languages[0]['href']"/>
```

Comment:
The canonical URL has nothing to do with the languages, it could very well be like your website has 2 domains (xxx.odoo.com vs mywebsite.be) and we need to show the real end point to avoid multiple indexation or to simply let crawler know which is the real canonical URL.
I am not sure why this is part of the `else` condition of the `if` condition checking language related stuff

(Also, for tracking purpose: note that if you are on a translated version of the website, the canonical is not the main lang one but the translated one on which you are, which is still working fine)